src/s3/commands/createBucket.ts

```diff
@@ -46,15 +44,12 @@ export async function createBucketCommand(
         window.showInformationMessage(localize('AWS.s3.createBucket.success', 'Created bucket: {0}', bucketName))
         telemetry.recordS3CreateBucket({ result: 'Succeeded' })
     } catch (e) {
-        getLogger().error(`Failed to create bucket ${bucketName}: %O`, e)
-        showViewLogsMessage(
-            localize('AWS.s3.createBucket.error.general', 'Failed to create bucket: {0}', bucketName),
-            window
-        )
+        const message = localize('AWS.s3.createBucket.error.general', 'Failed to create bucket: {0}', bucketName)
         telemetry.recordS3CreateBucket({ result: 'Failed' })
+        throw ToolkitError.chain(e, message)
+    } finally {
+        await refreshNode(node, commands)
```

Comment:
All command implementations should be refactored like this. It's just very tedious because some human attention is required even though most implementations are duplicated. For example, refreshing the node had to be moved to a `finally` block to keep the same behavior. At least this new pattern is easy to work with in terms of refactoring tools if that time ever comes.